
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Finding 16s/18s reads in metagenomes - bio(logist | informatician)</title>
  <meta name="author" content="Connor Skennerton">
  <link rel="author" href="humans.txt">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

  
    
  
  <meta name="description" content=" Got a Metagenome? want to know what the community looks like? ">
  
  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://ctSkennerton.github.io/blog/2014/04/19/rRNA-read-extraction-from-metagenomes/">
  <link href="/favicon.png" rel="icon">
  <link href='http://fonts.googleapis.com/css?family=Cantarell' rel='stylesheet' type='text/css'>
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/atom.xml" rel="alternate" title="bio(logist | informatician)" type="application/atom+xml">
  <meta name="og:type" content="website" />
  <meta name="og:site_name" content="bio(logist | informatician)" />
  <meta name="og:title" content="Finding 16s/18s reads in metagenomes" />
  <meta name="og:description" content=" Got a Metagenome? want to know what the community looks like? " />
  <meta name="og:url" content="http://ctSkennerton.github.io/blog/2014/04/19/rRNA-read-extraction-from-metagenomes/"/>
  <meta name="url" content="http://ctSkennerton.github.io/blog/2014/04/19/rRNA-read-extraction-from-metagenomes/">
  
  <meta name="subject" content="bioinformaticsmetagenomicstutorial"/>
  <meta name="category" content="bioinformaticsmetagenomicstutorial"/>
  
  <meta name="distribution" content="global">
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="http://fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="http://fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  

</head>

<body   >
  <nav role="navigation"><div class="navbar">
  <div class="navbar-inner">
    <a class="brand" href="/">bio(logist | informatician)</a>
    <ul class="nav">
      <li><a href="/">Home</a></li>
      <li><a href="/blog/archives">Archives</a></li>
    </ul>
    <ul class="nav" data-subscription="rss">
      <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
      
    </ul>
      
    <form class="navbar-form" action="http://google.com/search" method="get">
      <fieldset role="search">
        <input type="hidden" name="q" value="site:ctSkennerton.github.io" />
        <input class="span2" type="text" name="q" results="0" placeholder="Search"/>
      </fieldset>
    </form>
      
    
  </div>
</div>
</nav>
  <div class="wrapper_single">
    <div class="container">
      <article class="span8 offset2" role="article">
        <div class="article-format">

  <header>
    
      <h1 class="entry-title">Finding 16s/18s Reads in Metagenomes</h1>
    
    
      <p class="meta">
        
  


  
    <span class="byline author vcard">by <span class="fn">Connor Skennerton</span></span>
  

 - 
        








  


<time datetime="2014-04-19T22:51:00-07:00" pubdate data-updated="true">Apr 19<span>th</span>, 2014</time> - 
        

posted in
<span class="categories">
  
    <a class='category' href='/blog/categories/bioinformatics/'>bioinformatics</a>, <a class='category' href='/blog/categories/metagenomics/'>metagenomics</a>, <a class='category' href='/blog/categories/tutorial/'>tutorial</a>
  
</span>


        
      </p>
    
  </header>


  <div class="entry-content"><p>Got a Metagenome? want to know what the community looks like?</p>

<!-- more -->


<p>rRNA operons are typically poorly assembled in metagenomic datasets due to
highly conserved sequences.  More targeted assembly approaches may be necessary
to obtain accurate reconstructions from short read datasets. There are a few
ways in which we can extract reads originating from either 16S or 18S reads and
there are a number of programs (<a href="http://selab.janelia.org/software/ssu-align/">SSU-ALIGN</a>,
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21887657">rRNASelector</a>,
<a href="http://ribopicker.sourceforge.net/">riboPicker</a>,
<a href="http://bioinfo.lifl.fr/RNA/sortmerna/">SortMeRNA</a>,
blast, <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">bowtie</a> or <a href="https://github.com/lh3/bwa">bwa</a>) to name a few.</p>

<p>There are a other ways of looking at the communty from raw metagenomic
reads. These are mostly kmer-based approaches like
<a href="http://ccb.jhu.edu/software/kraken/">kraken</a> or extended marker gene
approaches like
<a href="http://huttenhower.sph.harvard.edu/metaphlan">MetaPhlAn</a> or
<a href="http://phylosift.wordpress.com/">phylosift</a>. I&rsquo;m not against any of
them, just never used them, so I&rsquo;m going to keep the following post
specific to 16s/18s community composition</p>

<h2>Extracting 16S/18S reads</h2>

<p>I&rsquo;ve used SortMeRNA and bowtie/bwa in my workflows with good success. The
difference between these two methods is that SortMeRNA that uses a kmer
searching method using an index created from a database of previously sequenced
genes whereas bowtie/bwa use a local alignment method to compare the query sequence to
a previously made database. Below are instructions on how to use both methods.</p>

<h2>Installing and using SortMeRNA</h2>

<p>From the source page for SortMeRNA on <a href="https://github.com/biocore/sortmerna">github</a>
copy the ssh clone url and then open up a terminal window and type</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$ </span>git clone git@github.com:biocore/sortmerna.git
</span><span class='line'><span class="nv">$ </span><span class="nb">cd </span>sortmerna
</span><span class='line'><span class="nv">$ </span>./configure --prefix<span class="o">=</span><span class="nv">$PWD</span>
</span><span class='line'><span class="nv">$ </span>make install
</span></code></pre></td></tr></table></div></figure>


<p><strong>Mac OSX install note:</strong> I found that there was an error with my install
with <code>configure</code> complaining that it was missing <code>install-sh</code>. I solved this by
copying <a href="http://sourceforge.net/projects/buildconf/">autogen.sh</a> into the same
directory as <code>configure</code>, execute it and then try running configure again.</p>

<p>The commands above should install SortMeRNA in the directory which you downloaded
it to.</p>

<h4>Building indexes for SortMeRNA</h4>

<p>SortMeRNA comes packaged with 8 different rRNA databases, all derived from SILVA.
Build either the 16S or 18S database, depending on what you want. Below is the
command used to generate the 18S index (assuming that you&rsquo;ve installed in the
download directory)</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$ </span>bin/indexdb_rna --ref rRNA_databases/silva-euk-18s-database-id95.fasta,index/silva-euk-18s-database-id95 --sensitive
</span></code></pre></td></tr></table></div></figure>


<p>If you <code>ls</code> the index directory you should see four files generated.</p>

<h4>Preprocessing reads</h4>

<p>Now we can use this index to extract the reads that may come from 18S using the
<code>sortmerna</code> command. I&rsquo;m going to assume that there are files containing raw
reads from an Illumina run called <code>file_R1.fastq.gz</code> and <code>file_R2.fastq.gz</code>.
Unfortunately one of the limitations of SortMeRNA is that it requires that you
only give it a single file and that the file is uncompressed. So to start with,
unzip the files using <code>gzip</code> and then combine them into a single file using
<code>merge-paired-reads.sh</code> found in the scripts directory of the SortMeRNA source
code</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$ </span>gunzip file_R1.fastq.gz file_R2.fastq.gz
</span><span class='line'><span class="nv">$ </span>bash scripts/merge-paired-reads.sh file_R1.fastq file_R2.fastq combined.fastq
</span></code></pre></td></tr></table></div></figure>


<p>To save space, it&rsquo;s probably best to re-zip the files to save on harddrive space</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$ </span>gzip file_R1.fastq file_R2.fastq
</span></code></pre></td></tr></table></div></figure>


<h4>Extracting the reads</h4>

<p>Now we can run <code>sortmerna</code>, saving matched reads (and their mates) to a file
with the prefix &lsquo;matched-18S&rsquo; in fastq format using 4 processors.</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$ </span>bin/sortmerna --reads combined.fastq <span class="se">\</span>
</span><span class='line'>    --ref rRNA_databases/silva-euk-18s-database-id95.fasta,index/silva-euk-18s-database-id95 <span class="se">\</span>
</span><span class='line'>    --paired-in <span class="se">\</span>
</span><span class='line'>    --fastx <span class="se">\</span>
</span><span class='line'>    --aligned matched-18S <span class="se">\</span>
</span><span class='line'>    -a 4
</span></code></pre></td></tr></table></div></figure>


<h2>Installing and using BWA</h2>

<p>My personal preference is to use bwa over bowtie but the merits of either are
debatable (there are also 50 other programs out there that try to solve the same
problem so the choice is yours). Download bwa from <a href="https://github.com/lh3/bwa">github</a>
by copying the ssh clone url and typing the following into the terminal. (Change
directories out of the SortMeRNA source directory before you do this)</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$ </span>git clone git@github.com:lh3/bwa.git
</span><span class='line'><span class="nv">$ </span><span class="nb">cd </span>bwa
</span><span class='line'><span class="nv">$ </span>make
</span></code></pre></td></tr></table></div></figure>


<p>bwa also requires that an index of the database be made. For simplicity, lets
use the same database that was included in SortMeRNA. To make the bwa index
copy the files that you want from the SortMeRNA source directory into a new
location.</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$ </span>cp ../sortmerna/rRNA_databases/silva-euk-18s-database-id95.fasta .
</span><span class='line'><span class="nv">$ </span>bwa index silva-euk-18s-database-id95.fasta
</span></code></pre></td></tr></table></div></figure>


<p>This should make a whole bunch of files that have extra file extensions appended
to the fasta file. With the index created we can now align the reads</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$ </span>bwa mem silva-euk-18s-database-id95.fasta file_R1.fastq.gz file_R2.fastq.gz &gt; aligned_18S.sam
</span></code></pre></td></tr></table></div></figure>


<p>(Notice that there is no preprocessing steps necessary for the reads)</p>

<h4>Postprocessing the sam file</h4>

<p>The output of bwa is a standardized format called <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">sam</a>
which many programs will now output. This format essentially describes the
alignment of each of the query sequences to the reference sequences. This is
not exactly what we want, which is the reads that were successful hits to the
reference in fasta/q format. To go from a sam file to a fasta/q file is a little
complicated (I wish it wasn&rsquo;t). To start with, download <a href="https://github.com/samtools/samtools">samtools</a>
and <a href="https://github.com/ctSkennerton/fxtract">fxtract</a> from github, and download
them into new source directories (bonus points for getting them installed without
a walkthrough). First convert the sam file into its equivalent binary format and
filter out unaligned sequences:</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$ </span>samtools view -SubF 4 aligned_18S.sam | samtools sort - aligned_18S <span class="o">&amp;&amp;</span> samtools index aligned_18S.bam
</span></code></pre></td></tr></table></div></figure>


<p>Now get the names of all the reads that aligned:</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$ </span>samtools view aligned_18S.bam | cut -f 1 &gt;aligned_reads.txt
</span></code></pre></td></tr></table></div></figure>


<p>And finally extract those reads from the original fastq files:</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'><span class="nv">$ </span>fxtract -Hf aligned_reads.txt file_R1.fastq.gz file_R2.fastq.gz &gt;aligned_reads.fastq
</span></code></pre></td></tr></table></div></figure>

</div>


  <footer>
    <p class="meta">
      
  


  
    <span class="byline author vcard">by <span class="fn">Connor Skennerton</span></span>
  

 - 
      








  


<time datetime="2014-04-19T22:51:00-07:00" pubdate data-updated="true">Apr 19<span>th</span>, 2014</time> - 
      

posted in
<span class="categories">
  
    <a class='category' href='/blog/categories/bioinformatics/'>bioinformatics</a>, <a class='category' href='/blog/categories/metagenomics/'>metagenomics</a>, <a class='category' href='/blog/categories/tutorial/'>tutorial</a>
  
</span>


    </p>
    
      <a class="pull-left" href="/blog/2013/12/27/poking-around-inside-grep/" title="Previous Post: Poking around inside grep">&laquo; Poking around inside grep</a>
    
    
      <a class="pull-right" href="/blog/2014/06/23/My-first-useful-script/" title="Next Post: My First Useful Script">My First Useful Script &raquo;</a>
    
  </footer>

</div>

        
      </article>
    </div>
  </div>
  <div id="footer-widgets">
  <div class="container">
    <div class="row">
  <div class="span3">
    <h2>recent posts</h2>
    <ul class="recent_posts">
      
        <li>
          <a href="/blog/2015/10/03/Bin-Decontamination/">Genome Bin Decontamination</a>
        </li>
      
        <li>
          <a href="/blog/2015/06/17/The-pace-of-genome-binning-from-metagenomics/">The pace of genome binning from metagenomes</a>
        </li>
      
        <li>
          <a href="/blog/2015/01/29/Genome-bin-completeness-and-contamination/">Genome Bin Completeness and Contamination</a>
        </li>
      
        <li>
          <a href="/blog/2014/11/12/Uploading-your-Theis-Lit-Review-to-Wikipedia/">Uploading your Theis Literature Review to Wikipedia</a>
        </li>
      
        <li>
          <a href="/blog/2014/06/23/My-first-useful-script/">My First Useful Script</a>
        </li>
      
    </ul>
    <h2><a href="/blog/archives">archives</a></h2>
  </div>
  <div class="span6">
    
<h2>twitter</h2>
<a class="twitter-timeline" href="https://twitter.com/ctskennerton" data-widget-id="460500926650793986">Tweets by @ctskennerton</a>
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+"://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>
<div class="tweet" data-twitter-user="ctskennerton">
</div>




  </div>
  <div class="span3">
    <h2>found on</h2>

<a href="https://github.com/ctSkennerton/" rel="tooltip" title="Github"><img class="social_icon" title="Github" alt="github icon" src="/images/glyphicons_381_github.png"></a>



<a href="http://stackoverflow.com/users/2049147/" rel="tooltip" title="Stack Overflow"><img class="social_icon" title="Stack Overflow" alt="stack overflow icon" src="/images/glyphicons_400_stack_overflow.png"></a>




<a href="http://twitter.com/ctskennerton" rel="tooltip" title="Twitter"><img class="social_icon" title="Twitter" alt="Twitter icon" src="/images/glyphicons_391_twitter_t.png"></a>







<a href="http://biostars.org/u/7059/" rel="tooltip" title="Biostar"><img class="social_icon" title="Biostar" alt="Biostar icon" src="/images/glyphicons_049_star.png"></a>


<a href="http://scholar.google.com.au/citations?user=ziBFFD4AAAAJ&hl=en" rel="tooltip" title="Google Scholar"><img class="social_icon" title="Google Scholar" alt="Google Scholar icon" src="/images/google_scholar.png"></a>



  </div>
</div>

  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-left">
  <a href="/">bio(logist | informatician)</a>
  - Copyright &copy; 2015 - Connor Skennerton
</p>
<p class="pull-right">
  <span>Powered by <a href="http://octopress.org/">Octopress</a>.</span>
  
    <span>Designed by <a href="http://www.AdrianArtiles.com">Adrian Artiles</a>.</span>
  
</p>

  </div>
</footer>

  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.7.2/jquery.min.js" type="text/javascript"></script>
<script>window.jQuery || document.write('<script src="/javascripts/libs/jquery-1.7.2.min.js" type="text/javascript"><\/script>')</script>
<script src="/javascripts/libs/bootstrap.min.js" type="text/javascript"></script>
<script src="/javascripts/jquery.tweet.js" type="text/javascript"></script>
<script src="/javascripts/jquery.instagram.js" type="text/javascript"></script>
<script src="/javascripts/libs/jquery.masonry.min.js" type="text/javascript"></script>
<script src="/javascripts/custom.js" type="text/javascript"></script>








  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = 'http://platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>




</body>
</html>
