<!DOCTYPE html>
<html lang="en-us">
<head>
  <meta charset="utf-8" />
  <title>Résumé | Connor Skennerton</title>
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="description" content="Experience (Senior/Staff) Data Engineer Pendulum Therapeutics 2019 — Present  Developed internal web applications using PostgreSQL or Amazon Neptune as the database layer, Flask/Django as the web server/API layer and VueJS frontend layer. Developed Python libraries so data scientists could easily interact with internal REST APIs and databases. Developed data analysis pipelines using Luigi for processing genomics data for genome assembly, variant calling, contamination detection, and metabolic reconstruction in AWS cloud infrastructure." />
  <meta name="author" content="Connor Skennerton" />
  <meta name="generator" content="Hugo 0.74.1" />
  <link href="/index.xml" rel="alternate" type="application/rss+xml" title="Connor Skennerton Feed" />
  <link rel="stylesheet" href="/style.4499e350b4bf6f8c081a7e09030944de5a9d76cb513b6ad32a95ea610fed2c95.css" />
  <link rel="stylesheet" href="/custom.css" />
  <script defer src="/script.9df38729991a6ec08d83b60514e841529cebd5ccc800cf8e1f98fb63e73aaa73.js"></script>
</head>
<body>
<div class="pure-g">
  <div class="pure-u-1-24 pure-u-md-5-24"></div>
  <div class="pure-u-22-24 pure-u-md-14-24">
  
<div class="navigation">
  <div class="navigation-header clearfix">
    <div class="pure-menu pure-menu-horizontal">
      <a class="pure-menu-heading pure-menu-link" href="/">
        Connor Skennerton
      </a>
      <ul class="pure-menu-list navigation-header-subtitle pull-end">
        <li class="pure-menu-item pure-menu-disabled">bio(logist | informatician)</li>
      </ul>
    </div>
  </div>
  <div class="navigation-content">
    <div class="pure-menu pure-menu-horizontal">
      <ul class="pure-menu-list">
        <li class="pure-menu-item" title="All posts">
          <a  class="pure-menu-link" href="/posts/">Posts</a>
        </li>
        <li class="pure-menu-item" title="About">
          <a  class="pure-menu-link" href="/about/">About</a>
        </li>
        <li class="pure-menu-item" title="Résumé">
          <a  class="pure-menu-link" href="/resume/">Résumé</a>
        </li>
      </ul>
    </div>
  </div>
</div>

  
<div>
  <div>
    <h2 class="post-title">
      
      Résumé
    </h2>
    
<div class="post-meta">
  <span>Date</span> &#x5b;
    <time datetime="2021-10-05T00:00:00Z">
      05 Oct 2021
    </time>
  &#x5d;
</div>

  </div>
  <div>
    <h2 id="experience">Experience</h2>
<h3 id="seniorstaff-data-engineer">(Senior/Staff) Data Engineer</h3>
<h4 id="pendulum-therapeutics-2019present">Pendulum Therapeutics 2019 — Present</h4>
<ul>
<li>Developed internal web applications using PostgreSQL or Amazon Neptune as the database layer, Flask/Django as the web server/API layer and VueJS frontend layer.</li>
<li>Developed Python libraries so data scientists could easily interact with internal REST APIs and databases.</li>
<li>Developed data analysis pipelines using Luigi for processing genomics data for genome assembly, variant calling, contamination detection, and metabolic reconstruction in AWS cloud infrastructure.</li>
</ul>
<h3 id="bioinformatics-cloud-engineer">Bioinformatics Cloud Engineer</h3>
<h4 id="second-genome-2018--2019">Second Genome 2018 — 2019</h4>
<ul>
<li>Converted bioinformatics pipelines for processing genomic and metagenomic data into serverless, containerized applications using Docker, AWS Batch, AWS Lambda, AWS SQS.</li>
<li>Spearheaded python code consolidation within the company into a single unified library, enforcing test-driven development and software development lifecycle practices.</li>
<li>Developed ETL pipelines to automatically scrape NCBI and IMG databases and integrate them with private datasets into data warehouses</li>
</ul>
<h3 id="postdoctoral-research-fellow">Postdoctoral Research Fellow</h3>
<h4 id="caltech-20142018">Caltech 2014―2018</h4>
<p>My role was the head bioinformatician for a diverse group of microbiologists, chemists and data scientists at the California Institute of Technology (Caltech). My day to day tasks included the processing of genomic and proteomic data, and generating analysis summaries and plots using statistical software and programing tools such as R and python. As part of this role I was responsible for the production of automated pipelines for data analysis and have developed numerous tools and scripts for processing genomic data. I implemented SQL databases to store the extensive genomic and proteomic data that were generated and developed simple interfaces such that other members of the group could access this information without in-depth knowledge of SQL.</p>
<h3 id="graduate-student">Graduate Student</h3>
<h4 id="university-of-quensland-20102014">University of Quensland 2010―2014</h4>
<p>My PhD research focused on metagenomic analysis of microbial and viral communities from engineered ecosystems found in wastewater treatment plants. Throughout my PhD I was at the interface of microbial ecology, chemical engineering, and computational biology and have gained many skills ranging from microscopy, reactor operation and computer programming. I developed a number of bioinformatic pipelines and substantial computer software in perl, python and C++. My background in molecular biology allowed me to place the inferences of metagenomic data into a broader biological framework and work with other members of the lab that did not have computer skills.</p>
<h2 id="education">Education</h2>
<h3 id="university-of-queensland">University of Queensland</h3>
<p>2014, PhD in Microbial Ecology</p>
<p>2010, Bachelor of Biotechnology</p>
<h3 id="honors--awards">Honors &amp; Awards</h3>
<p>2015, Dean&rsquo;s award for outstanding PhD dissertation, University of Queensland</p>
<p>2010, Australian Research Council PhD Fellowship</p>
<h2 id="publications">Publications</h2>
<p>McMurdie, P. J., Stoeva, M. K., Justice, N., Nemchek, M., Sieber, C. M. K., Tyagi, S., Gines J., <strong>Skennerton, C. T.,</strong> Souza, M., Kolterman, O., Eid, J. (2022). &ldquo;<a href="https://doi.org/10.1186/s12866-021-02415-8">Increased circulating butyrate and ursodeoxycholate during probiotic intervention in humans with type 2 diabetes</a>&rdquo;. BMC Microbiol 22, 19 (2022).</p>
<p>Chadwick, G. L., <strong>Skennerton, C. T.,</strong> Laso-Pérez, R., Leu, A. O., Speth, D. R., Yu, H., Morgan-Lang, C., Hatzenpichler, R., Goudeau, D., Malmstrom, R., Brazelton, W. J., Woyke, T., Hallam, S. J., Tyson, G. W., Wegener, G., Boetius, A., Orphan, V. J. (2022) &ldquo;<a href="https://doi.org/10.1371/journal.pbio.3001508">Comparative genomics reveals electron transfer and syntrophic mechanisms differentiating methanotrophic and methanogenic archaea</a>&rdquo;. PLoS Biol 20(1): e3001508.</p>
<p>Yu, H., <strong>Skennerton, C. T.,</strong> Chadwick, G. L., Leu, A. O., Aoki, M., Tyson, G. W., Orphan, V. J. (2022) &ldquo;<a href="https://pubmed.ncbi.nlm.nih.gov/34285362/">Sulfate differentially stimulates but is not respired by diverse anaerobic methanotrophic archaea</a>&rdquo;. ISME J 16(1): 168-177.</p>
<p>Sim, M., <strong>Skennerton, C. T.,</strong> Orphan, V. J. (2021) &ldquo;<a href="https://pubmed.ncbi.nlm.nih.gov/33444327/">Physiological, genomic, and sulfur isotopic characterization of methanol metabolism by Desulfovibrio carbinolicus</a>&rdquo;. PLoS ONE. 16(1): e0245069</p>
<p>Yu, H., Susanti, D., McGlynn, S. E., <strong>Skennerton, C. T.,</strong> Chourey, K., Iyer, R., Scheller, S., Tavormina, P. L., Hettich, R. L., Mukhopadhyay, B., Orphan V. J. (2018) &ldquo;<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6286981/">Comparative Genomics and Proteomic Analysis of Assimilatory Sulfate Reduction Pathways in Anaerobic Methanotrophic Archaea</a>&rdquo;. Frontiers Microbiol. 9: 2917</p>
<p><strong>Skennerton, C. T.</strong>, Chourey, K., Iyer, R., Hettich, R. L., Tyson, G. W., Orphan, V. J. (2017) &ldquo;<a href="https://www.ncbi.nlm.nih.gov/pubmed/28765215">Methane-Fueled Syntrophy through Extracellular Electron Transfer: Uncovering the Genomic Traits Conserved within Diverse Bacterial Partners of Anaerobic Methanotrophic Archaea</a>&rdquo;. mBio 8(4):e00530-17</p>
<p><strong>Skennerton, C. T.</strong>, Haroon, M. F., Briegel, A., Shi, J., Jensen, G. J., Tyson, G. W., Orphan, V. J. (2016) &ldquo;<a href="http://www.ncbi.nlm.nih.gov/pubmed/27058507">Phylogenomic analysis of <em>Candidatus</em> ‘Izimaplasma’ species: free-living representatives from a Tenericutes clade found in methane seeps</a>&rdquo;. ISMEJ. 10: 2679–2692</p>
<p>Marlow, J. J., <strong>Skennerton, C. T.,</strong> Li, Z., Chourey, K., Hettich, R. L., Pan, C., Orphan, V. J. (2016). &ldquo;<a href="http://journal.frontiersin.org/article/10.3389/fmicb.2016.00563/full">Proteomic Stable Isotope Probing Reveals Biosynthesis Dynamics of Slow Growing Methane Based Microbial Communities</a>&rdquo;. Frontiers Microbiol. 7: 563</p>
<p>Barr, J. J., Dutilh, B. E., <strong>Skennerton, C. T.</strong>, Fukushima, T., Hastie, M. L., Gorman, J. J., Tyson, G. W., Bond, P. L. (2016). “<a href="http://www.ncbi.nlm.nih.gov/pubmed/26279094">Metagenomic and metaproteomic analyses of Accumulibacter phosphatis enriched floccular and granular biofilm</a>”. Environ. Microbiol. 18(1): 273–287</p>
<p><strong>Skennerton, C. T.,</strong> Ward, L. M., Michel, A., Metcalfe, K., Valiente, C., Mullin, S., Chan, K. Y., Gradinaru, V., Orphan, V. J. (2015). &ldquo;<a href="http://journal.frontiersin.org/article/10.3389/fmicb.2015.01425/full">Genomic Reconstruction of an Uncultured Hydrothermal Vent Gammaproteobacterial Methanotroph (Family <em>Methylothermaceae</em>) Indicates Multiple Adaptations to Oxygen Limitation</a>&quot;. Frontiers Microbiol. 6</p>
<p>Parks, D. H., Imelfort, M., <strong>Skennerton, C. T.</strong>, Hugenholtz, P., Tyson, G. W. (2015). “<a href="http://genome.cshlp.org/content/25/7/1043.long">CheckM: assessing the quality of microbial genomes recovered from isolates, single cells, and metagenomes</a>”. Genome Res. 25: 1043–1055.</p>
<p><strong>Skennerton, C. T.</strong>, Barr, J. J., Slater, F. R., Bond, P. L., Tyson, G. W. (2015). “<a href="http://www.ncbi.nlm.nih.gov/pubmed/25088527">Expanding our view of genomic diversity in <em>Candidatus</em> Accumulibacter clades</a>”. Environ. Microbiol. 17 (5): 1574–1585.</p>
<p>Soo, R. M., <strong>Skennerton, C. T.</strong>, Sekiguchi, Y., Imelfort, M., Paech, S. J., Dennis, P. G., Steen, J. A., Parks, D. H., Tyson, G. W., Hugenholtz, P. (2014). “<a href="http://gbe.oxfordjournals.org/content/6/5/1031.long">An expanded genomic representation of the phylum cyanobacteria</a>”. Genome Biol. Evol. 6 (5): 1031– 1045.</p>
<p>Haroon, M. F., <strong>Skennerton, C. T.</strong>, Steen, J. A., Lachner, N., Hugenholtz, P., Tyson, G. W. (2013). “<a href="http://www.ncbi.nlm.nih.gov/pubmed/24060113">In-Solution Fluorescence In Situ Hybridization and Fluorescence-Activated Cell Sorting for Single Cell and Population Genome Recovery</a>”. In: Microbial Metagenomics, Metatranscriptomics, and Metaproteomics. Ed. by Edward F. DeLong. Vol. 531. Academic Press, 3–19.</p>
<p><strong>Skennerton, C. T.</strong>, Imelfort, M., Tyson, G. W. (2013). “<a href="http://nar.oxfordjournals.org/content/41/10/e105.long">Crass: identification and reconstruction of CRISPR from unassembled metagenomic data</a>”. Nucleic Acids Res. 41 (10): e105.</p>
<p><strong>Skennerton, C. T.</strong>, Angly, F. E., Breitbart, M., Bragg, L., He, S., McMahon, K. D., Hugenholtz, P., Tyson, G. W. (2011) “<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0020095">Phage encoded H-NS: a potential achilles heel in the bacterial defence system</a>”. PLoS One, 6 (5): e20095.</p>

  </div>
</div>
  <div class="footer">
  <div class="pure-menu pure-menu-horizontal">
    <ul class="pure-menu-list">
      <li class="pure-menu-item" title="Theme repository">
        <a class="pure-menu-link" href="https://github.com/ctSkennerton">GitHub</a>
      </li>
      <li class="pure-menu-item" title="RSS Feed">
        <a href="/index.xml" class="pure-menu-link">RSS</a>
      </li>
      <li class="pure-menu-item fix-cursor-pointer" title="Go to top">
        <a class="pure-menu-link" id="btn-gototop">
          <span class="fix-placement-up">&#x21e7;&#xfe0e;</span>
        </a>
      </li>
    </ul>
  </div>
  <div class="pure-menu pure-menu-horizontal">
    <ul class="pure-menu-list">
      <li class="pure-menu-item pure-menu-disabled">
        &copy; 2022 &mdash; Connor Skennerton — All rights reserved.
      </li>
    </ul>
  </div>
</div>

  </div>
  <div class="pure-u-1-24 pure-u-md-5-24"></div>
</div>
</body>
</html>
