<!DOCTYPE html>
<html lang="en-us">
<head>
  <meta charset="utf-8" />
  <title>Connor Skennerton</title>
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="description" content="Connor Skennerton&#39;s personal website" />
  <meta name="author" content="Connor Skennerton" />
  <meta name="generator" content="Hugo 0.74.1" />
  <link href="/index.xml" rel="alternate" type="application/rss+xml" title="Connor Skennerton Feed" />
  <link rel="stylesheet" href="/style.4499e350b4bf6f8c081a7e09030944de5a9d76cb513b6ad32a95ea610fed2c95.css" />
  <link rel="stylesheet" href="/custom.css" />
  <script defer src="/script.9df38729991a6ec08d83b60514e841529cebd5ccc800cf8e1f98fb63e73aaa73.js"></script>
</head>
<body>
<div class="pure-g">
  <div class="pure-u-1-24 pure-u-md-5-24"></div>
  <div class="pure-u-22-24 pure-u-md-14-24">
  
<div class="navigation">
  <div class="navigation-header clearfix">
    <div class="pure-menu pure-menu-horizontal">
      <a class="pure-menu-heading pure-menu-link" href="/">
        Connor Skennerton
      </a>
      <ul class="pure-menu-list navigation-header-subtitle pull-end">
        <li class="pure-menu-item pure-menu-disabled">bio(logist | informatician)</li>
      </ul>
    </div>
  </div>
  <div class="navigation-content">
    <div class="pure-menu pure-menu-horizontal">
      <ul class="pure-menu-list">
        <li class="pure-menu-item" title="All posts">
          <a  class="pure-menu-link" href="/posts/">Posts</a>
        </li>
        <li class="pure-menu-item" title="About">
          <a  class="pure-menu-link" href="/about/">About</a>
        </li>
        <li class="pure-menu-item" title="Résumé">
          <a  class="pure-menu-link" href="/resume/">Résumé</a>
        </li>
      </ul>
    </div>
  </div>
</div>

  
<div>
  <div>
    <h2 class="post-title">
      
      <a href="/2019/10/01/an-update-on-bioawk/">An update on Bioawk</a>
    </h2>
    
<div class="post-meta">
  <span>Date</span> &#x5b;
    <time datetime="2019-10-01T17:26:23Z">
      01 Oct 2019
    </time>
  &#x5d;
        
        
  <span>Tags</span> &#x5b;
    <a href="/tags/bioinformatics">Bioinformatics</a>
  &#x5d;
</div>

  </div>
  <div>
    Bioawk is a great project started by Heng Li some years ago. The aim was to take the awk source code and modify it slightly for use with common biological formats and adding in some new functions. Heng&rsquo;s original doesn&rsquo;t accept too many pull requests so to add in some features, I maintain my own fork that has a few improvements.
Long ago I added in a translate function and recently I added in a function that will take the attribute string from a GFF file and turn it into an awk array.
  </div>
</div>
<div>
  <div>
    <h2 class="post-title">
      
      <a href="/2018/04/10/getting-data-from-ncbi-assembly-using-the-accession-number-only/">Getting data from NCBI assembly using the accession number only</a>
    </h2>
    
<div class="post-meta">
  <span>Date</span> &#x5b;
    <time datetime="2018-04-10T23:18:11Z">
      10 Apr 2018
    </time>
  &#x5d;
        
        
  <span>Tags</span> &#x5b;
    <a href="/tags/bioinformatics">Bioinformatics</a>
  &#x5d;
</div>

  </div>
  <div>
    NCBI&rsquo;s assembly database is a great one-stop-shop for genomic data and annotations but it&rsquo;s actually kind of difficult to download data if you only know the accession number of an assembly. The documentation says that the assembly database is integrated with entrez-direct, a great set of command line utilities for accessing NCBI data from the command line. Most of the databases have an option to download data based on the ID, so I thought that something like the following would work
  </div>
</div>
<div>
  <div>
    <h2 class="post-title">
      
      <a href="/2017/12/15/creating-huge-metabolic-overviews-for-comparative-genomics/">Creating huge metabolic overviews for comparative genomics</a>
    </h2>
    
<div class="post-meta">
  <span>Date</span> &#x5b;
    <time datetime="2017-12-15T19:38:46Z">
      15 Dec 2017
    </time>
  &#x5d;
        
        
  <span>Tags</span> &#x5b;
    <a href="/tags/bioinformatics">Bioinformatics</a>
  &#x5d;
</div>

  </div>
  <div>
    I love looking at KEGG maps and using them to understand an organisms metabolism but they have their limitations. For starters, you&rsquo;re obviously stuck with how they are drawn, which in most cases includes many variations on a particular pathway. Secondly, the tools for mapping on your own genes to a pathway are limited to one organism at a time. What I really wanted was a way of quickly comparing 10s of genomes across a common set of pathways, in this case amino acid synthesis pathways.
  </div>
</div>
<div>
  <div>
    <h2 class="post-title">
      
      <a href="/2017/07/22/formatting-scientific-names-in-r/">Formatting scientific names in R</a>
    </h2>
    
<div class="post-meta">
  <span>Date</span> &#x5b;
    <time datetime="2017-07-22T17:27:02Z">
      22 Jul 2017
    </time>
  &#x5d;
        
        
  <span>Tags</span> &#x5b;
    <a href="/tags/bioinformatics">Bioinformatics</a>
    <a href="/tags/r">R</a>
  &#x5d;
</div>

  </div>
  <div>
    <p>Here is a function that will take a character string in R and return an expression for fancy formatting in plots that properly italicize scientific names. The syntax for doing this is truly quite horrible, but this is how R does it.</p>
  </div>
</div>
<div>
  <div>
    <h2 class="post-title">
      
      <a href="/2017/04/09/split_by-splitting-files-on-the-command-line-based-on-content/">split_by: splitting files on the command line based on content</a>
    </h2>
    
<div class="post-meta">
  <span>Date</span> &#x5b;
    <time datetime="2017-04-09T08:04:41Z">
      09 Apr 2017
    </time>
  &#x5d;
        
        
  <span>Tags</span> &#x5b;
    <a href="/tags/bioinformatics">Bioinformatics</a>
  &#x5d;
</div>

  </div>
  <div>
    Unix has so many great ways to perform text manipulation but one niche which hasn&rsquo;t been filled is splitting a tabular file into pieces based on the contents of certain columns. There are two commands, split and csplit, that do a similar role. split can split a file into a certain number of bytes or lines; csplit uses a regular expression to determine where to split the file. Often for my purposes neither of these tools is a good fit, and what I really want is an equivalent to the &ldquo;group by&rdquo; clause in SQL databases.
  </div>
</div>
<div>
  <div>
    <h2 class="post-title">
      
      <a href="/2016/11/20/downloading-ncbi-genomes-from-a-given-taxonomy/">Downloading NCBI genomes from a given taxonomy</a>
    </h2>
    
<div class="post-meta">
  <span>Date</span> &#x5b;
    <time datetime="2016-11-20T05:18:43Z">
      20 Nov 2016
    </time>
  &#x5d;
        
        
  <span>Tags</span> &#x5b;
    <a href="/tags/bioinformatics">Bioinformatics</a>
  &#x5d;
</div>

  </div>
  <div>
    <p>The <a href="https://www.ncbi.nlm.nih.gov/books/NBK179288/">Entrez Direct</a> toolkit is great for programmatic access to all of NCBI&rsquo;s resources. This little snippet below finds all of the refseq representative genomes for a given NCBI taxonomy ID, makes a little summary of the genomes downloaded and uses <code>wget</code> to download the genbank files from the Assembly FTP. Change the inital query in the first call to <code>esearch</code> to change what genomes are downloaded.</p>
  </div>
</div>
<div>
  <div>
    <h2 class="post-title">
      
      <a href="/2016/11/17/python-ete3-formatting-organism-names-the-way-i-want/">Python ete3: formatting organism names the way I want</a>
    </h2>
    
<div class="post-meta">
  <span>Date</span> &#x5b;
    <time datetime="2016-11-17T00:58:40Z">
      17 Nov 2016
    </time>
  &#x5d;
        
        
  <span>Tags</span> &#x5b;
    <a href="/tags/bioinformatics">Bioinformatics</a>
  &#x5d;
</div>

  </div>
  <div>
    I feel like I&rsquo;m on a life-long quest to make all of my phylogenetic tree figures completely programmatically. The best tool I&rsquo;ve found for making them is the ete library for the python programing language. I&rsquo;ve already figured out how to get trees drawn in the style that I like but there was still one thing left to do: making organism names italicize correctly. I work with microorganisms where the convention is for the genus and species names to be italicized but the strain name to be in regular font.
  </div>
</div>
<div>
  <div>
    <h2 class="post-title">
      
      <a href="/2016/09/22/my-guide-to-annotating-proteins-and-pathways/">My guide to annotating proteins and pathways</a>
    </h2>
    
<div class="post-meta">
  <span>Date</span> &#x5b;
    <time datetime="2016-09-22T22:14:50Z">
      22 Sep 2016
    </time>
  &#x5d;
        
        
  <span>Tags</span> &#x5b;
    <a href="/tags/bioinformatics">Bioinformatics</a>
  &#x5d;
</div>

  </div>
  <div>
    So. You&rsquo;ve got yourself a nice new genome sequence and you want to know what kind of metabolism it has. There is a good chance that you have some idea already — you think it&rsquo;s a nitrogen fixer or a sulfate reducer etc. — based on other analyses and now it&rsquo;s time to strengthen your paper with a bit of genomic evidence.
Getting an initial annotation The vast majority of the genes in the genome are going to be hypothetical proteins; of the rest, a sizable chunk are going to be genes with a general sort of annotation like &ldquo;ABC transporter&rdquo; (which says nothing about what it&rsquo;s transporting), and the rest are going to be metabolic genes that you probably care about.
  </div>
</div>
<div>
  <div>
    <h2 class="post-title">
      
      <a href="/2016/08/12/messing-around-with-phyloxml-in-ete3/">Messing around with Phyloxml in ete3</a>
    </h2>
    
<div class="post-meta">
  <span>Date</span> &#x5b;
    <time datetime="2016-08-12T22:09:04Z">
      12 Aug 2016
    </time>
  &#x5d;
        
        
  <span>Tags</span> &#x5b;
    <a href="/tags/bioinformatics">Bioinformatics</a>
  &#x5d;
</div>

  </div>
  <div>
    ete3 has support for phyloxml which I use with archaeopteryx tree viewer for a lot of my day-to-day phylogenetics visualisation. My main reason for using phyloxml is one of convenience as I have a script that will easily add in the proper organism name onto the tree and I think that archaeopteryx is a really good basic tree viewer. I wanted to draw a tree from phyloxml in ete using my own style and to have the proper organism name to be rendered.
  </div>
</div>
<div>
  <div>
    <h2 class="post-title">
      
      <a href="/2016/07/11/going-from-a-messy-supplementary-table-to-good-clean-data/">Going from a messy supplementary table to good clean data</a>
    </h2>
    
<div class="post-meta">
  <span>Date</span> &#x5b;
    <time datetime="2016-07-11T21:00:53Z">
      11 Jul 2016
    </time>
  &#x5d;
        
        
  <span>Tags</span> &#x5b;
    <a href="/tags/bioinformatics">Bioinformatics</a>
    <a href="/tags/r">R</a>
    <a href="/tags/tidy-data">tidy data</a>
  &#x5d;
</div>

  </div>
  <div>
    Bioinformatics&hellip; Or &lsquo;advanced file copying&rsquo; as I like to call it.
— Nick Loman (@pathogenomenick) January 29, 2014
  Get ready for some advanced file copying! I recently had to clean up some data from the supplementary material from Pereira et. al 2011, which is a very nice table of manually annotated genes in sulfate reducing bacteria. The only problem is that the table is designed for maximum human readability, which made it a real pain when trying to parse out the data.
  </div>
</div>

  <div class="pagination-content">
    <div class="pure-menu pure-menu-horizontal">
      <ul class="pure-menu-list">
        <li class="pure-menu-item pure-menu-disabled">
          Pages
        </li>
        <li class="pure-menu-item">
          <a href="/" class="pure-menu-link">First</a>
        </li>
        <li class="pure-menu-item pure-menu-selected">
          <a href="/page/2/" class="pure-menu-link">2</a>
        </li>
        <li class="pure-menu-item">
          <a href="/page/3/" class="pure-menu-link">Last</a>
        </li>
      </ul>
    </div>
  </div>

  <div class="footer">
  <div class="pure-menu pure-menu-horizontal">
    <ul class="pure-menu-list">
      <li class="pure-menu-item" title="Theme repository">
        <a class="pure-menu-link" href="https://github.com/ctSkennerton">GitHub</a>
      </li>
      <li class="pure-menu-item" title="RSS Feed">
        <a href="/index.xml" class="pure-menu-link">RSS</a>
      </li>
      <li class="pure-menu-item fix-cursor-pointer" title="Go to top">
        <a class="pure-menu-link" id="btn-gototop">
          <span class="fix-placement-up">&#x21e7;&#xfe0e;</span>
        </a>
      </li>
    </ul>
  </div>
  <div class="pure-menu pure-menu-horizontal">
    <ul class="pure-menu-list">
      <li class="pure-menu-item pure-menu-disabled">
        &copy; 2022 &mdash; Connor Skennerton — All rights reserved.
      </li>
    </ul>
  </div>
</div>

  </div>
  <div class="pure-u-1-24 pure-u-md-5-24"></div>
</div>
</body>
</html>
