
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Genome Bin Decontamination - bio(logist | informatician)</title>
  <meta name="author" content="Connor Skennerton">
  <link rel="author" href="humans.txt">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

  
    
  
  <meta name="description" content="Genome bins comming off automated pipelines can be contaminated
with parts of other genomes. As part of my workflow I use
CheckM
(I&rsquo;m biased &hellip;">
  
  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://ctSkennerton.github.io/blog/2015/10/03/Bin-Decontamination/">
  <link href="/favicon.png" rel="icon">
  <link href='http://fonts.googleapis.com/css?family=Cantarell' rel='stylesheet' type='text/css'>
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/atom.xml" rel="alternate" title="bio(logist | informatician)" type="application/atom+xml">
  <meta name="og:type" content="website" />
  <meta name="og:site_name" content="bio(logist | informatician)" />
  <meta name="og:title" content="Genome Bin Decontamination" />
  <meta name="og:description" content="Genome bins comming off automated pipelines can be contaminated
with parts of other genomes. As part of my workflow I use
CheckM
(I&rsquo;m biased &hellip;" />
  <meta name="og:url" content="http://ctSkennerton.github.io/blog/2015/10/03/Bin-Decontamination/"/>
  <meta name="url" content="http://ctSkennerton.github.io/blog/2015/10/03/Bin-Decontamination/">
  
  <meta name="subject" content="bioinformatics"/>
  <meta name="category" content="bioinformatics"/>
  
  <meta name="distribution" content="global">
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="http://fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="http://fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  

</head>

<body   >
  <nav role="navigation"><div class="navbar">
  <div class="navbar-inner">
    <a class="brand" href="/">bio(logist | informatician)</a>
    <ul class="nav">
      <li><a href="/">Home</a></li>
      <li><a href="/blog/archives">Archives</a></li>
    </ul>
    <ul class="nav" data-subscription="rss">
      <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
      
    </ul>
      
    <form class="navbar-form" action="http://google.com/search" method="get">
      <fieldset role="search">
        <input type="hidden" name="q" value="site:ctSkennerton.github.io" />
        <input class="span2" type="text" name="q" results="0" placeholder="Search"/>
      </fieldset>
    </form>
      
    
  </div>
</div>
</nav>
  <div class="wrapper_single">
    <div class="container">
      <article class="span8 offset2" role="article">
        <div class="article-format">

  <header>
    
      <h1 class="entry-title">Genome Bin Decontamination</h1>
    
    
      <p class="meta">
        
  


  
    <span class="byline author vcard">by <span class="fn">Connor Skennerton</span></span>
  

 - 
        








  


<time datetime="2015-10-03T00:00:00-07:00" pubdate data-updated="true">Oct 3<span>rd</span>, 2015</time> - 
        

posted in
<span class="categories">
  
    <a class='category' href='/blog/categories/bioinformatics/'>bioinformatics</a>
  
</span>


        
      </p>
    
  </header>


  <div class="entry-content"><p>Genome bins comming off automated pipelines can be contaminated
with parts of other genomes. As part of my workflow I use
<a href="http://genome.cshlp.org/content/early/2015/05/14/gr.186072.114.abstract">CheckM</a>
(I&rsquo;m biased since I&rsquo;m a coauthor) to assess the contamination of
genome bins using single-copy marker genes. If you&rsquo;re lucky then
the genome bins that you&rsquo;re interested in will be relatively complete
without much contamination. Unfortunately that isn&rsquo;t always the
case. In this blog post I&rsquo;m going to run through some of the analyses
that I did on a genome bin that was 90% complete but 70% contaminated.
This is exploratory analysis to see if I can manually improve the
bin over what the automated tools can do.</p>

<!--more-->


<h2>Background</h2>

<p>I&rsquo;m not going into details on how I got this bin but briefly I had
three metagenomic samples, named 3730, 5133, and 5579, which were
assembled with
<a href="http://bioinformatics.oxfordjournals.org/content/31/10/1674.long">megahit</a>
and binned using <a href="https://peerj.com/articles/1165/">metabat</a>. I ran
the genome bins from sample 3730 through CheckM. In the below analysis I&rsquo;m going to
try to improve <strong>bin_41</strong>.</p>

<h2>Analysis</h2>

<h3>Creating the input files</h3>

<p>For starters I want the <code>.depth.txt</code> file created by metabat, which I copied and renamed <code>3730_coverage.tsv</code>. Next I created a mapping file for which contigs belonged to which bins. I have the fasta files of all of the bins in a separate directory. To get the mapping of the fasta files I ran the following:</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='sh'><span class='line'>grep -oP <span class="s1">&#39;(?&lt;=^\&gt;)\S+&#39;</span> *fa | awk <span class="s1">&#39;BEGIN{FS=&quot;:&quot;;OFS=&quot;\t&quot;}{print $2,$1}&#39;</span> &gt; 3730_bin_mapping.tsv
</span></code></pre></td></tr></table></div></figure>


<p>And finally I wanted to know where all of the multi-copy markers were so I created a file based on the CheckM output, with a bit of reformatting in awk:</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='sh'><span class='line'>checkm qa -o 6 ckm_results/lineage.ms ckm_results | awk <span class="s1">&#39;BEGIN{FS=&quot;\t&quot;;OFS=&quot;,&quot;}{n = split($3,a,&quot;,&quot;);for(i = 1; i &lt;= n; ++i){print $1,$2,a[i]}}&#39;</span> &gt; 3730_multiple_markers.csv
</span></code></pre></td></tr></table></div></figure>


<h3>Exploratory analysis in R</h3>

<div>
  <pre><code class='r'>library(readr)
duplicated_markers = read_csv(&quot;3730_multiple_markers.csv&quot;)
coverage = read_tsv(&quot;3730_coverage.tsv&quot;)
# no header in the file, so give the columns names
bin_mapping = read_tsv(&quot;3730_bin_mapping.tsv&quot;, col_names = c(&quot;contigName&quot;, &quot;bin&quot;))</code></pre>
</div>


<p>Clean up the dataframes to make the column names consistent and remove a bit of unneeded text. Note that these commands are specific to how files were named on my system, you may not need to do this or change this section to meet your own needs.</p>

<div>
  <pre><code class='r'>bin_mapping$bin = gsub(&quot;metabat_binned/final.contigs.fa.metabat_&quot;, &quot;&quot;, bin_mapping$bin)
bin_mapping$bin = gsub(&quot;\\.fa$&quot;, &quot;&quot;, bin_mapping$bin)
duplicated_markers$`Bin Id` = gsub(&quot;final.contigs.fa.metabat_&quot;, &quot;&quot;, duplicated_markers$`Bin Id`)
duplicated_markers$`Gene Ids` = gsub(&quot;_\\d+$&quot;, &quot;&quot;, duplicated_markers$`Gene Ids`, perl=TRUE)
colnames(duplicated_markers) &lt;- c(&quot;bin&quot;, &quot;marker&quot;, &quot;contigName&quot;)</code></pre>
</div>


<p>Create a smaller dataframe of just the binned contigs</p>

<div>
  <pre><code class='r'>binned_contigs = merge(bin_mapping, coverage, by=&quot;contigName&quot;)</code></pre>
</div>


<p>We are interested in Bin 41, lets get just the data for it.</p>

<div>
  <pre><code class='r'>bin_41 = binned_contigs[binned_contigs$bin == &quot;bins_41&quot;,]</code></pre>
</div>


<p>Since we have three samples we can look at the coverage of the contigs in each sample as a matrix of 2D</p>

<div>
  <pre><code class='r'>pairs(~final.contigs.3730.bam+final.contigs.5133.bam+final.contigs.5579.bam, data=bin_41, main = &quot;Comparison of contig coverage between samples&quot;, labels = c(&quot;3730&quot;, &quot;5133&quot;, &quot;5579&quot;))</code></pre>
</div>


<p><img src="/images/bin-decontamination-paired-coverage-1.svg" alt="plot of chunk bin-decontamination-paired-coverage" />
The majority of the contigs are found in sample 3730 between 10-15x and with very low coverage in sample 5133. Some of the contigs also have some coverage in sample 5579, but most don&rsquo;t. It&rsquo;s tempting to remove all of the contigs that don&rsquo;t fit into the band of coverage, but from this we can&rsquo;t be certain if the duplicated markers are in these outlier contigs.</p>

<p>Since the coverage of sample 5133 doesn&rsquo;t really factor into things lets look at a single 2D plot of the 3730 coverage and the 5579 coverage. As a coarse approach lets take a look at where all of the contigs with duplicated markers are in this plot. For starters make a new column in the bin_41 dataframe that tells us if the contig contains any duplicated marker.</p>

<div>
  <pre><code class='r'>bin_41_duplicated_markers = subset(duplicated_markers, bin == &quot;bins_41&quot;)
bin_41$containsDuplicateMarker &lt;- bin_41$contigName %in% unique(bin_41_duplicated_markers$contigName)</code></pre>
</div>


<p>And now lets plot the data, changing the color of the points based on the value of the &lsquo;containsDuplicateMarker&rsquo; column</p>

<div>
  <pre><code class='r'>plot(bin_41$final.contigs.3730.bam, bin_41$final.contigs.5579.bam, pch = 19, col = &quot;lightgrey&quot;, cex=0.5, xlab=&quot;3730 coverage&quot;, ylab=&quot;5579 coverage&quot;, main=&quot;Contigs containing duplicated markers&quot;)
points(bin_41[bin_41$containsDuplicateMarker,]$final.contigs.3730.bam, bin_41[bin_41$containsDuplicateMarker,]$final.contigs.5579.bam, col = &quot;red&quot;)</code></pre>
</div>


<p><img src="/images/bin-decontamination-all-duplicate-markers-1.svg" alt="plot of chunk bin-decontamination-all-duplicate-markers" />
Apart from a couple of outliers, the majority of the contigs that contain multiple markers are in the central mass of contigs. There doesn&rsquo;t appear to be any way to systematically remove a substantial amount of contamination in this genome bin.</p>

<p>Instead of looking at all contigs that contain any duplicated markers, we can also visualise the positions of contigs for a specific duplicated marker. You can get a list of the markers that are duplicated in the bin by using the unique command. Then create a logical vector of bin_41 of the contigs that contain any of the particular markers.</p>

<div>
  <pre><code class='r'>unique(bin_41_duplicated_markers$marker)</code></pre>
</div>




<div>
  <pre><code class='text'>##   [1] &quot;PF03439.8&quot;  &quot;PF01172.13&quot; &quot;PF00164.20&quot; &quot;PF09173.6&quot;  &quot;PF03874.11&quot;
##   [6] &quot;PF00203.16&quot; &quot;PF00466.15&quot; &quot;PF00416.17&quot; &quot;PF05670.8&quot;  &quot;PF04127.10&quot;
##  [11] &quot;PF01090.14&quot; &quot;PF01193.19&quot; &quot;PF09249.6&quot;  &quot;PF11987.3&quot;  &quot;PF00298.14&quot;
##  [16] &quot;PF00252.13&quot; &quot;PF01198.14&quot; &quot;PF01201.17&quot; &quot;PF01351.13&quot; &quot;PF04567.12&quot;
##  [21] &quot;PF01922.12&quot; &quot;PF01667.12&quot; &quot;PF00398.15&quot; &quot;TIGR03685&quot;  &quot;PF04983.13&quot;
##  [26] &quot;TIGR03683&quot;  &quot;PF04561.9&quot;  &quot;PF04565.11&quot; &quot;PF04919.7&quot;  &quot;TIGR00425&quot; 
##  [31] &quot;PF01200.13&quot; &quot;PF07541.7&quot;  &quot;PF00679.19&quot; &quot;PF01194.12&quot; &quot;PF13656.1&quot; 
##  [36] &quot;PF01015.13&quot; &quot;PF01282.14&quot; &quot;PF01287.15&quot; &quot;PF00276.15&quot; &quot;PF01780.14&quot;
##  [41] &quot;TIGR00064&quot;  &quot;TIGR00389&quot;  &quot;PF04997.7&quot;  &quot;PF04563.10&quot; &quot;PF03947.13&quot;
##  [46] &quot;PF01912.13&quot; &quot;PF00181.18&quot; &quot;PF04010.8&quot;  &quot;TIGR00549&quot;  &quot;TIGR01018&quot; 
##  [51] &quot;PF01157.13&quot; &quot;TIGR00392&quot;  &quot;PF00347.18&quot; &quot;PF00831.18&quot; &quot;TIGR01080&quot; 
##  [56] &quot;TIGR00408&quot;  &quot;TIGR00270&quot;  &quot;PF08068.7&quot;  &quot;PF00687.16&quot; &quot;PF04560.15&quot;
##  [61] &quot;PF05000.12&quot; &quot;PF00380.14&quot; &quot;PF00752.12&quot; &quot;PF09377.5&quot;  &quot;TIGR02389&quot; 
##  [66] &quot;TIGR00419&quot;  &quot;PF01000.21&quot; &quot;PF04566.8&quot;  &quot;PF00562.23&quot; &quot;PF00366.15&quot;
##  [71] &quot;TIGR00057&quot;  &quot;PF01191.14&quot; &quot;PF00572.13&quot; &quot;PF01981.11&quot; &quot;PF01984.15&quot;
##  [76] &quot;TIGR03677&quot;  &quot;PF00411.14&quot; &quot;PF01864.12&quot; &quot;PF00238.14&quot; &quot;TIGR03724&quot; 
##  [81] &quot;PF00177.16&quot; &quot;PF00827.12&quot; &quot;TIGR02338&quot;  &quot;PF01849.13&quot; &quot;TIGR00468&quot; 
##  [86] &quot;PF01866.12&quot; &quot;PF01982.11&quot; &quot;PF00237.14&quot; &quot;PF03876.12&quot; &quot;PF02978.14&quot;
##  [91] &quot;TIGR00670&quot;  &quot;PF00623.15&quot; &quot;PF01246.15&quot; &quot;PF00573.17&quot; &quot;PF00189.15&quot;
##  [96] &quot;TIGR01213&quot;  &quot;TIGR00344&quot;  &quot;PF01868.11&quot; &quot;PF03764.13&quot; &quot;PF00867.13&quot;</code></pre>
</div>




<div>
  <pre><code class='r'>contains_x_marker = bin_41$contigName %in% subset(bin_41_duplicated_markers, marker == &quot;PF01157.13&quot;)$contigName</code></pre>
</div>


<p>With this vector we can plot the points like we did with the complete set of duplicated markers.</p>

<div>
  <pre><code class='r'>plot(bin_41$final.contigs.3730.bam, bin_41$final.contigs.5579.bam, pch = 19, col = &quot;lightgrey&quot;, cex=0.5, main=&quot;Position of contigs with PF01157.13&quot;, xlab=&quot;3730 coverage&quot;, ylab=&quot;5579 coverage&quot;)
points(bin_41[contains_x_marker,]$final.contigs.3730.bam, bin_41[contains_x_marker,]$final.contigs.5579.bam, col = &quot;red&quot;)</code></pre>
</div>


<p><img src="/images/bin-decontamination-single-marker-positions-1.svg" alt="plot of chunk bin-decontamination-single-marker-positions" /></p>

<p>So now we can see that ther are three contigs containing this marker, one
appears to be an outlier, but the other two contigs have quite similar
coverage profiles</p>
</div>


  <footer>
    <p class="meta">
      
  


  
    <span class="byline author vcard">by <span class="fn">Connor Skennerton</span></span>
  

 - 
      








  


<time datetime="2015-10-03T00:00:00-07:00" pubdate data-updated="true">Oct 3<span>rd</span>, 2015</time> - 
      

posted in
<span class="categories">
  
    <a class='category' href='/blog/categories/bioinformatics/'>bioinformatics</a>
  
</span>


    </p>
    
      <a class="pull-left" href="/blog/2015/06/17/The-pace-of-genome-binning-from-metagenomics/" title="Previous Post: The pace of genome binning from metagenomes">&laquo; The pace of genome binning from metagenomes</a>
    
    
  </footer>

</div>

        
      </article>
    </div>
  </div>
  <div id="footer-widgets">
  <div class="container">
    <div class="row">
  <div class="span3">
    <h2>recent posts</h2>
    <ul class="recent_posts">
      
        <li>
          <a href="/blog/2015/10/03/Bin-Decontamination/">Genome Bin Decontamination</a>
        </li>
      
        <li>
          <a href="/blog/2015/06/17/The-pace-of-genome-binning-from-metagenomics/">The pace of genome binning from metagenomes</a>
        </li>
      
        <li>
          <a href="/blog/2015/01/29/Genome-bin-completeness-and-contamination/">Genome Bin Completeness and Contamination</a>
        </li>
      
        <li>
          <a href="/blog/2014/11/12/Uploading-your-Theis-Lit-Review-to-Wikipedia/">Uploading your Theis Literature Review to Wikipedia</a>
        </li>
      
        <li>
          <a href="/blog/2014/06/23/My-first-useful-script/">My First Useful Script</a>
        </li>
      
    </ul>
    <h2><a href="/blog/archives">archives</a></h2>
  </div>
  <div class="span6">
    
<h2>twitter</h2>
<a class="twitter-timeline" href="https://twitter.com/ctskennerton" data-widget-id="460500926650793986">Tweets by @ctskennerton</a>
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+"://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>
<div class="tweet" data-twitter-user="ctskennerton">
</div>




  </div>
  <div class="span3">
    <h2>found on</h2>

<a href="https://github.com/ctSkennerton/" rel="tooltip" title="Github"><img class="social_icon" title="Github" alt="github icon" src="/images/glyphicons_381_github.png"></a>



<a href="http://stackoverflow.com/users/2049147/" rel="tooltip" title="Stack Overflow"><img class="social_icon" title="Stack Overflow" alt="stack overflow icon" src="/images/glyphicons_400_stack_overflow.png"></a>




<a href="http://twitter.com/ctskennerton" rel="tooltip" title="Twitter"><img class="social_icon" title="Twitter" alt="Twitter icon" src="/images/glyphicons_391_twitter_t.png"></a>







<a href="http://biostars.org/u/7059/" rel="tooltip" title="Biostar"><img class="social_icon" title="Biostar" alt="Biostar icon" src="/images/glyphicons_049_star.png"></a>


<a href="http://scholar.google.com.au/citations?user=ziBFFD4AAAAJ&hl=en" rel="tooltip" title="Google Scholar"><img class="social_icon" title="Google Scholar" alt="Google Scholar icon" src="/images/google_scholar.png"></a>



  </div>
</div>

  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-left">
  <a href="/">bio(logist | informatician)</a>
  - Copyright &copy; 2015 - Connor Skennerton
</p>
<p class="pull-right">
  <span>Powered by <a href="http://octopress.org/">Octopress</a>.</span>
  
    <span>Designed by <a href="http://www.AdrianArtiles.com">Adrian Artiles</a>.</span>
  
</p>

  </div>
</footer>

  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.7.2/jquery.min.js" type="text/javascript"></script>
<script>window.jQuery || document.write('<script src="/javascripts/libs/jquery-1.7.2.min.js" type="text/javascript"><\/script>')</script>
<script src="/javascripts/libs/bootstrap.min.js" type="text/javascript"></script>
<script src="/javascripts/jquery.tweet.js" type="text/javascript"></script>
<script src="/javascripts/jquery.instagram.js" type="text/javascript"></script>
<script src="/javascripts/libs/jquery.masonry.min.js" type="text/javascript"></script>
<script src="/javascripts/custom.js" type="text/javascript"></script>








  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = 'http://platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>




</body>
</html>
