<!DOCTYPE html>
<html lang="en-us">
<head>
  <meta charset="utf-8" />
  <title>Genome Bin Decontamination | Connor Skennerton</title>
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="description" content="Genome bins comming off automated pipelines can be contaminated
with parts of other genomes. As part of my workflow I use
CheckM
(I&rsquo;m biased since I&rsquo;m a coauthor) to assess the contamination of
genome bins using single-copy marker genes. If you&rsquo;re lucky then
the genome bins that you&rsquo;re interested in will be relatively complete
without much contamination. Unfortunately that isn&rsquo;t always the
case. In this blog post I&rsquo;m going to run through some of the analyses
that I did on a genome bin that was 90% complete but 70% contaminated.
This is exploratory analysis to see if I can manually improve the
bin over what the automated tools can do." />
  <meta name="author" content="Connor Skennerton" />
  <meta name="generator" content="Hugo 0.74.1" />
  <link href="/index.xml" rel="alternate" type="application/rss+xml" title="Connor Skennerton Feed" />
  <link rel="stylesheet" href="/style.7c6d960b20273ed88bc63a60591a57e7739fb21f243b77f27c3bc730ef6205c0.css" />
  <link rel="stylesheet" href="/custom.css" />
  <script defer src="/script.9df38729991a6ec08d83b60514e841529cebd5ccc800cf8e1f98fb63e73aaa73.js"></script>
</head>
<body>
<div class="pure-g">
  <div class="pure-u-1-24 pure-u-md-5-24"></div>
  <div class="pure-u-22-24 pure-u-md-14-24">
  
<div class="navigation">
  <div class="navigation-header clearfix">
    <div class="pure-menu pure-menu-horizontal">
      <a class="pure-menu-heading pure-menu-link" href="/">
        Connor Skennerton
      </a>
      <ul class="pure-menu-list navigation-header-subtitle pull-end">
        <li class="pure-menu-item pure-menu-disabled">bio(logist | informatician)</li>
      </ul>
    </div>
  </div>
  <div class="navigation-content">
    <div class="pure-menu pure-menu-horizontal">
      <ul class="pure-menu-list">
        <li class="pure-menu-item" title="All posts">
          <a  class="pure-menu-link" href="/posts/">Posts</a>
        </li>
        <li class="pure-menu-item" title="About">
          <a  class="pure-menu-link" href="/about/">About</a>
        </li>
        <li class="pure-menu-item" title="Résumé">
          <a  class="pure-menu-link" href="/resume/">Résumé</a>
        </li>
      </ul>
    </div>
  </div>
</div>

  
<div>
  <div>
    <h2 class="post-title">
      
      Genome Bin Decontamination
    </h2>
    
<div class="post-meta">
  <span>Date</span> &#x5b;
    <time datetime="2015-10-03T00:00:00Z">
      03 Oct 2015
    </time>
  &#x5d;
        
        
  <span>Tags</span> &#x5b;
    <a href="/tags/bioinformatics">bioinformatics</a>
  &#x5d;
</div>

  </div>
  <div>
    <p>Genome bins comming off automated pipelines can be contaminated
with parts of other genomes. As part of my workflow I use
<a href="http://genome.cshlp.org/content/early/2015/05/14/gr.186072.114.abstract">CheckM</a>
(I&rsquo;m biased since I&rsquo;m a coauthor) to assess the contamination of
genome bins using single-copy marker genes. If you&rsquo;re lucky then
the genome bins that you&rsquo;re interested in will be relatively complete
without much contamination. Unfortunately that isn&rsquo;t always the
case. In this blog post I&rsquo;m going to run through some of the analyses
that I did on a genome bin that was 90% complete but 70% contaminated.
This is exploratory analysis to see if I can manually improve the
bin over what the automated tools can do.</p>
<h2 id="background">Background</h2>
<p>I&rsquo;m not going into details on how I got this bin but briefly I had
three metagenomic samples, named 3730, 5133, and 5579, which were
assembled with
<a href="http://bioinformatics.oxfordjournals.org/content/31/10/1674.long">megahit</a>
and binned using <a href="https://peerj.com/articles/1165/">metabat</a>. I ran
the genome bins from sample 3730 through CheckM. In the below analysis I&rsquo;m going to
try to improve <strong>bin_41</strong>.</p>
<h2 id="analysis">Analysis</h2>
<h3 id="creating-the-input-files">Creating the input files</h3>
<p>For starters I want the <code>.depth.txt</code> file created by metabat, which I copied and renamed <code>3730_coverage.tsv</code>. Next I created a mapping file for which contigs belonged to which bins. I have the fasta files of all of the bins in a separate directory. To get the mapping of the fasta files I ran the following:</p>
<div class="highlight"><pre class="chroma"><code class="language-sh" data-lang="sh">grep -oP <span class="s1">&#39;(?&lt;=^\&gt;)\S+&#39;</span> *fa <span class="p">|</span> awk <span class="s1">&#39;BEGIN{FS=&#34;:&#34;;OFS=&#34;\t&#34;}{print $2,$1}&#39;</span> &gt; 3730_bin_mapping.tsv
</code></pre></div><p>And finally I wanted to know where all of the multi-copy markers were so I created a file based on the CheckM output, with a bit of reformatting in awk:</p>
<div class="highlight"><pre class="chroma"><code class="language-sh" data-lang="sh">checkm qa -o <span class="m">6</span> ckm_results/lineage.ms ckm_results <span class="p">|</span> awk <span class="s1">&#39;BEGIN{FS=&#34;\t&#34;;OFS=&#34;,&#34;}{n = split($3,a,&#34;,&#34;);for(i = 1; i &lt;= n; ++i){print $1,$2,a[i]}}&#39;</span> &gt; 3730_multiple_markers.csv
</code></pre></div><h3 id="exploratory-analysis-in-r">Exploratory analysis in R</h3>
<div class="highlight"><pre class="chroma"><code class="language-R" data-lang="R"><span class="nf">library</span><span class="p">(</span><span class="n">readr</span><span class="p">)</span>
<span class="n">duplicated_markers</span> <span class="o">=</span> <span class="nf">read_csv</span><span class="p">(</span><span class="s">&#34;3730_multiple_markers.csv&#34;</span><span class="p">)</span>
<span class="n">coverage</span> <span class="o">=</span> <span class="nf">read_tsv</span><span class="p">(</span><span class="s">&#34;3730_coverage.tsv&#34;</span><span class="p">)</span>
<span class="c1"># no header in the file, so give the columns names</span>
<span class="n">bin_mapping</span> <span class="o">=</span> <span class="nf">read_tsv</span><span class="p">(</span><span class="s">&#34;3730_bin_mapping.tsv&#34;</span><span class="p">,</span> <span class="n">col_names</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;contigName&#34;</span><span class="p">,</span> <span class="s">&#34;bin&#34;</span><span class="p">))</span>
</code></pre></div><p>Clean up the dataframes to make the column names consistent and remove a bit of unneeded text. Note that these commands are specific to how files were named on my system, you may not need to do this or change this section to meet your own needs.</p>
<div class="highlight"><pre class="chroma"><code class="language-R" data-lang="R"><span class="n">bin_mapping</span><span class="o">$</span><span class="n">bin</span> <span class="o">=</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;metabat_binned/final.contigs.fa.metabat_&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="n">bin_mapping</span><span class="o">$</span><span class="n">bin</span><span class="p">)</span>
<span class="n">bin_mapping</span><span class="o">$</span><span class="n">bin</span> <span class="o">=</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;\\.fa$&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="n">bin_mapping</span><span class="o">$</span><span class="n">bin</span><span class="p">)</span>
<span class="n">duplicated_markers</span><span class="o">$</span><span class="n">`Bin Id`</span> <span class="o">=</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;final.contigs.fa.metabat_&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="n">duplicated_markers</span><span class="o">$</span><span class="n">`Bin Id`</span><span class="p">)</span>
<span class="n">duplicated_markers</span><span class="o">$</span><span class="n">`Gene Ids`</span> <span class="o">=</span> <span class="nf">gsub</span><span class="p">(</span><span class="s">&#34;_\\d+$&#34;</span><span class="p">,</span> <span class="s">&#34;&#34;</span><span class="p">,</span> <span class="n">duplicated_markers</span><span class="o">$</span><span class="n">`Gene Ids`</span><span class="p">,</span> <span class="n">perl</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">duplicated_markers</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;bin&#34;</span><span class="p">,</span> <span class="s">&#34;marker&#34;</span><span class="p">,</span> <span class="s">&#34;contigName&#34;</span><span class="p">)</span>
</code></pre></div><p>Create a smaller dataframe of just the binned contigs</p>
<div class="highlight"><pre class="chroma"><code class="language-R" data-lang="R"><span class="n">binned_contigs</span> <span class="o">=</span> <span class="nf">merge</span><span class="p">(</span><span class="n">bin_mapping</span><span class="p">,</span> <span class="n">coverage</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="s">&#34;contigName&#34;</span><span class="p">)</span>
</code></pre></div><p>We are interested in Bin 41, lets get just the data for it.</p>
<div class="highlight"><pre class="chroma"><code class="language-R" data-lang="R"><span class="n">bin_41</span> <span class="o">=</span> <span class="n">binned_contigs[binned_contigs</span><span class="o">$</span><span class="n">bin</span> <span class="o">==</span> <span class="s">&#34;bins_41&#34;</span><span class="p">,</span><span class="n">]</span>
</code></pre></div><p>Since we have three samples we can look at the coverage of the contigs in each sample as a matrix of 2D</p>
<div class="highlight"><pre class="chroma"><code class="language-R" data-lang="R"><span class="nf">pairs</span><span class="p">(</span><span class="o">~</span><span class="n">final.contigs.3730.bam</span><span class="o">+</span><span class="n">final.contigs.5133.bam</span><span class="o">+</span><span class="n">final.contigs.5579.bam</span><span class="p">,</span> <span class="n">data</span><span class="o">=</span><span class="n">bin_41</span><span class="p">,</span> <span class="n">main</span> <span class="o">=</span> <span class="s">&#34;Comparison of contig coverage between samples&#34;</span><span class="p">,</span> <span class="n">labels</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&#34;3730&#34;</span><span class="p">,</span> <span class="s">&#34;5133&#34;</span><span class="p">,</span> <span class="s">&#34;5579&#34;</span><span class="p">))</span>
</code></pre></div><p><img src="/media/bin-decontamination-paired-coverage-1.svg" alt="plot of chunk bin-decontamination-paired-coverage">
The majority of the contigs are found in sample 3730 between 10-15x and with very low coverage in sample 5133. Some of the contigs also have some coverage in sample 5579, but most don&rsquo;t. It&rsquo;s tempting to remove all of the contigs that don&rsquo;t fit into the band of coverage, but from this we can&rsquo;t be certain if the duplicated markers are in these outlier contigs.</p>
<p>Since the coverage of sample 5133 doesn&rsquo;t really factor into things lets look at a single 2D plot of the 3730 coverage and the 5579 coverage. As a coarse approach lets take a look at where all of the contigs with duplicated markers are in this plot. For starters make a new column in the bin_41 dataframe that tells us if the contig contains any duplicated marker.</p>
<div class="highlight"><pre class="chroma"><code class="language-R" data-lang="R"><span class="n">bin_41_duplicated_markers</span> <span class="o">=</span> <span class="nf">subset</span><span class="p">(</span><span class="n">duplicated_markers</span><span class="p">,</span> <span class="n">bin</span> <span class="o">==</span> <span class="s">&#34;bins_41&#34;</span><span class="p">)</span>
<span class="n">bin_41</span><span class="o">$</span><span class="n">containsDuplicateMarker</span> <span class="o">&lt;-</span> <span class="n">bin_41</span><span class="o">$</span><span class="n">contigName</span> <span class="o">%in%</span> <span class="nf">unique</span><span class="p">(</span><span class="n">bin_41_duplicated_markers</span><span class="o">$</span><span class="n">contigName</span><span class="p">)</span>
</code></pre></div><p>And now lets plot the data, changing the color of the points based on the value of the &lsquo;containsDuplicateMarker&rsquo; column</p>
<div class="highlight"><pre class="chroma"><code class="language-R" data-lang="R"><span class="nf">plot</span><span class="p">(</span><span class="n">bin_41</span><span class="o">$</span><span class="n">final.contigs.3730.bam</span><span class="p">,</span> <span class="n">bin_41</span><span class="o">$</span><span class="n">final.contigs.5579.bam</span><span class="p">,</span> <span class="n">pch</span> <span class="o">=</span> <span class="m">19</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="s">&#34;lightgrey&#34;</span><span class="p">,</span> <span class="n">cex</span><span class="o">=</span><span class="m">0.5</span><span class="p">,</span> <span class="n">xlab</span><span class="o">=</span><span class="s">&#34;3730 coverage&#34;</span><span class="p">,</span> <span class="n">ylab</span><span class="o">=</span><span class="s">&#34;5579 coverage&#34;</span><span class="p">,</span> <span class="n">main</span><span class="o">=</span><span class="s">&#34;Contigs containing duplicated markers&#34;</span><span class="p">)</span>
<span class="nf">points</span><span class="p">(</span><span class="n">bin_41[bin_41</span><span class="o">$</span><span class="n">containsDuplicateMarker</span><span class="p">,</span><span class="n">]</span><span class="o">$</span><span class="n">final.contigs.3730.bam</span><span class="p">,</span> <span class="n">bin_41[bin_41</span><span class="o">$</span><span class="n">containsDuplicateMarker</span><span class="p">,</span><span class="n">]</span><span class="o">$</span><span class="n">final.contigs.5579.bam</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="s">&#34;red&#34;</span><span class="p">)</span>
</code></pre></div><figure>
    <img src="/media/bin-decontamination-all-duplicate-markers-1.svg"/> 
</figure>
<p>Apart from a couple of outliers, the majority of the contigs that contain multiple markers are in the central mass of contigs. There doesn&rsquo;t appear to be any way to systematically remove a substantial amount of contamination in this genome bin.</p>
<p>Instead of looking at all contigs that contain any duplicated markers, we can also visualise the positions of contigs for a specific duplicated marker. You can get a list of the markers that are duplicated in the bin by using the unique command. Then create a logical vector of bin_41 of the contigs that contain any of the particular markers.</p>
<div class="highlight"><pre class="chroma"><code class="language-R" data-lang="R"><span class="nf">unique</span><span class="p">(</span><span class="n">bin_41_duplicated_markers</span><span class="o">$</span><span class="n">marker</span><span class="p">)</span>

<span class="c1">##   [1] &#34;PF03439.8&#34;  &#34;PF01172.13&#34; &#34;PF00164.20&#34; &#34;PF09173.6&#34;  &#34;PF03874.11&#34;</span>
<span class="c1">##   [6] &#34;PF00203.16&#34; &#34;PF00466.15&#34; &#34;PF00416.17&#34; &#34;PF05670.8&#34;  &#34;PF04127.10&#34;</span>
<span class="c1">##  [11] &#34;PF01090.14&#34; &#34;PF01193.19&#34; &#34;PF09249.6&#34;  &#34;PF11987.3&#34;  &#34;PF00298.14&#34;</span>
<span class="c1">##  [16] &#34;PF00252.13&#34; &#34;PF01198.14&#34; &#34;PF01201.17&#34; &#34;PF01351.13&#34; &#34;PF04567.12&#34;</span>
<span class="c1">##  [21] &#34;PF01922.12&#34; &#34;PF01667.12&#34; &#34;PF00398.15&#34; &#34;TIGR03685&#34;  &#34;PF04983.13&#34;</span>
<span class="c1">##  [26] &#34;TIGR03683&#34;  &#34;PF04561.9&#34;  &#34;PF04565.11&#34; &#34;PF04919.7&#34;  &#34;TIGR00425&#34; </span>
<span class="c1">##  [31] &#34;PF01200.13&#34; &#34;PF07541.7&#34;  &#34;PF00679.19&#34; &#34;PF01194.12&#34; &#34;PF13656.1&#34; </span>
<span class="c1">##  [36] &#34;PF01015.13&#34; &#34;PF01282.14&#34; &#34;PF01287.15&#34; &#34;PF00276.15&#34; &#34;PF01780.14&#34;</span>
<span class="c1">##  [41] &#34;TIGR00064&#34;  &#34;TIGR00389&#34;  &#34;PF04997.7&#34;  &#34;PF04563.10&#34; &#34;PF03947.13&#34;</span>
<span class="c1">##  [46] &#34;PF01912.13&#34; &#34;PF00181.18&#34; &#34;PF04010.8&#34;  &#34;TIGR00549&#34;  &#34;TIGR01018&#34; </span>
<span class="c1">##  [51] &#34;PF01157.13&#34; &#34;TIGR00392&#34;  &#34;PF00347.18&#34; &#34;PF00831.18&#34; &#34;TIGR01080&#34; </span>
<span class="c1">##  [56] &#34;TIGR00408&#34;  &#34;TIGR00270&#34;  &#34;PF08068.7&#34;  &#34;PF00687.16&#34; &#34;PF04560.15&#34;</span>
<span class="c1">##  [61] &#34;PF05000.12&#34; &#34;PF00380.14&#34; &#34;PF00752.12&#34; &#34;PF09377.5&#34;  &#34;TIGR02389&#34; </span>
<span class="c1">##  [66] &#34;TIGR00419&#34;  &#34;PF01000.21&#34; &#34;PF04566.8&#34;  &#34;PF00562.23&#34; &#34;PF00366.15&#34;</span>
<span class="c1">##  [71] &#34;TIGR00057&#34;  &#34;PF01191.14&#34; &#34;PF00572.13&#34; &#34;PF01981.11&#34; &#34;PF01984.15&#34;</span>
<span class="c1">##  [76] &#34;TIGR03677&#34;  &#34;PF00411.14&#34; &#34;PF01864.12&#34; &#34;PF00238.14&#34; &#34;TIGR03724&#34; </span>
<span class="c1">##  [81] &#34;PF00177.16&#34; &#34;PF00827.12&#34; &#34;TIGR02338&#34;  &#34;PF01849.13&#34; &#34;TIGR00468&#34; </span>
<span class="c1">##  [86] &#34;PF01866.12&#34; &#34;PF01982.11&#34; &#34;PF00237.14&#34; &#34;PF03876.12&#34; &#34;PF02978.14&#34;</span>
<span class="c1">##  [91] &#34;TIGR00670&#34;  &#34;PF00623.15&#34; &#34;PF01246.15&#34; &#34;PF00573.17&#34; &#34;PF00189.15&#34;</span>
<span class="c1">##  [96] &#34;TIGR01213&#34;  &#34;TIGR00344&#34;  &#34;PF01868.11&#34; &#34;PF03764.13&#34; &#34;PF00867.13&#34;</span>

<span class="n">contains_x_marker</span> <span class="o">=</span> <span class="n">bin_41</span><span class="o">$</span><span class="n">contigName</span> <span class="o">%in%</span> <span class="nf">subset</span><span class="p">(</span><span class="n">bin_41_duplicated_markers</span><span class="p">,</span> <span class="n">marker</span> <span class="o">==</span> <span class="s">&#34;PF01157.13&#34;</span><span class="p">)</span><span class="o">$</span><span class="n">contigName</span>
</code></pre></div><p>With this vector we can plot the points like we did with the complete set of duplicated markers.</p>
<div class="highlight"><pre class="chroma"><code class="language-R" data-lang="R"><span class="nf">plot</span><span class="p">(</span><span class="n">bin_41</span><span class="o">$</span><span class="n">final.contigs.3730.bam</span><span class="p">,</span> <span class="n">bin_41</span><span class="o">$</span><span class="n">final.contigs.5579.bam</span><span class="p">,</span> <span class="n">pch</span> <span class="o">=</span> <span class="m">19</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="s">&#34;lightgrey&#34;</span><span class="p">,</span> <span class="n">cex</span><span class="o">=</span><span class="m">0.5</span><span class="p">,</span> <span class="n">main</span><span class="o">=</span><span class="s">&#34;Position of contigs with PF01157.13&#34;</span><span class="p">,</span> <span class="n">xlab</span><span class="o">=</span><span class="s">&#34;3730 coverage&#34;</span><span class="p">,</span> <span class="n">ylab</span><span class="o">=</span><span class="s">&#34;5579 coverage&#34;</span><span class="p">)</span>
<span class="nf">points</span><span class="p">(</span><span class="n">bin_41[contains_x_marker</span><span class="p">,</span><span class="n">]</span><span class="o">$</span><span class="n">final.contigs.3730.bam</span><span class="p">,</span> <span class="n">bin_41[contains_x_marker</span><span class="p">,</span><span class="n">]</span><span class="o">$</span><span class="n">final.contigs.5579.bam</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="s">&#34;red&#34;</span><span class="p">)</span>
</code></pre></div><figure>
    <img src="/media/bin-decontamination-single-marker-positions-1.svg"/> 
</figure>
<p>So now we can see that ther are three contigs containing this marker, one
appears to be an outlier, but the other two contigs have quite similar
coverage profiles</p>
  </div>
</div>
  <div class="footer">
  <div class="pure-menu pure-menu-horizontal">
    <ul class="pure-menu-list">
      <li class="pure-menu-item" title="Theme repository">
        <a class="pure-menu-link" href="https://github.com/ctSkennerton">GitHub</a>
      </li>
      <li class="pure-menu-item" title="RSS Feed">
        <a href="/index.xml" class="pure-menu-link">RSS</a>
      </li>
      <li class="pure-menu-item fix-cursor-pointer" title="Go to top">
        <a class="pure-menu-link" id="btn-gototop">
          <span class="fix-placement-up">&#x21e7;&#xfe0e;</span>
        </a>
      </li>
    </ul>
  </div>
  <div class="pure-menu pure-menu-horizontal">
    <ul class="pure-menu-list">
      <li class="pure-menu-item pure-menu-disabled">
        &copy; 2021 &mdash; Connor Skennerton — All rights reserved.
      </li>
    </ul>
  </div>
</div>

  </div>
  <div class="pure-u-1-24 pure-u-md-5-24"></div>
</div>
</body>
</html>
